File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def hanoi(n,start,via,to):
    global answer
    if n ==1:
        answer.append([start,to])
    else:
        hanoi(n-1, start, to ,via)
        answer.append([start,to])
        hanoi(n-1, via, start ,to)
        
def solution(n):
    global answer
    answer =[]
    hanoi(n, 1,2,3)
    return answer

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    var answer = [];
    function hanoi(n,start, via, end){
        if (n == 1) answer.push([start,end]);
        else{
            hanoi(n-1,start,end,via);
            answer.push([start,end]);
            hanoi(n-1,via,start,end);
        }
    }
    hanoi(n,1,2,3);
    return answer;
}